src/main/scala/net/snowflake/spark/snowflake/catalog/SfScan.scala

```diff
@@ -0,0 +1,35 @@
+package net.snowflake.spark.snowflake.catalog
+
+import net.snowflake.spark.snowflake.SnowflakeRelation
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.connector.read.V1Scan
+import org.apache.spark.sql.sources.{BaseRelation, Filter, TableScan}
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{Row, SQLContext}
+
+case class SfScan(
```

Comment:
Can you explain how `SfScan/SfScanBuilder/SfWriteBuilder/SfTable` are used.
They are defined in this PR but I don't understand when and how they are used?
Could you please add some test cases for them?

Reply:
OK